app/services/create_api_key.rb

```diff
@@ -0,0 +1,105 @@
+class CreateApiKey
+  CLIENT_ID_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".freeze
+  CLIENT_ID_LENGTH = 17
+  CLIENT_ID_PREFIX = "HUB".freeze
+  SECRET_LENGTH = 24
+  API_KEY_TYPE = "API_KEY".freeze
+  PLAN_LIST_LIMIT = (ENV["USAGE_PLAN_LIST_PAGINATION_LIMIT"] || "100").to_i
+  PER_FPO_RATE_LIMIT = (ENV["USAGE_PLAN_PER_FPO_RATE_LIMIT"] || "100").to_i
+  PER_FPO_BURST_LIMIT = (ENV["USAGE_PLAN_PER_FPO_BURST_LIMIT"] || "200").to_i
+  REST_API_ID = ENV["REST_API_ID"]
+  STAGE_NAME = ENV["REST_STAGE_NAME"]
+  TAGS = { customer: "fpo" }.freeze
+
+  def initialize(api_gateway_client = Aws::APIGateway::Client.new)
+    @api_gateway_client = api_gateway_client
+  end
+
+  def call(organisation_id, description = nil)
+    api_key = ApiKey.new
+    api_key.organisation_id = Organisation.find_by(organisation_id: organisation_id).id
+    api_key.api_key_id = generate_client_id
+    api_key.secret = generate_random_secret
+    api_key.enabled = true
+    api_key.description = description || "Autogenerated on #{Time.zone.now.utc.iso8601}"
+
+    create_in_api_gateway(api_key)
+    associate_to_usage_plan(api_key)
+
+    api_key.save!
```

Comment:
what happens if it fails to save? 👀 save would return false but this isn't caught somewhere right?

Reply:
I was just reviewing that part myself and thinking we should make it a bit more robust

Reply:
Added a fresh commit to fix that

Reply:
Lovely! I'm happy with it. Approving now!